src/Areas/Monitor/Commands/Metrics/MetricsNamespacesCommand.cs

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using AzureMcp.Areas.Monitor.Models;
+using AzureMcp.Areas.Monitor.Options;
+using AzureMcp.Areas.Monitor.Options.Metrics;
+using AzureMcp.Areas.Monitor.Services;
+using AzureMcp.Commands.Monitor;
+using Microsoft.Extensions.Logging;
+
+namespace AzureMcp.Areas.Monitor.Commands.Metrics;
+
+/// <summary>
+/// Command for listing Azure Monitor metric namespaces
+/// </summary>
+public sealed class MetricsNamespacesCommand(ILogger<MetricsNamespacesCommand> logger)
+    : BaseMetricsCommand<MetricsNamespacesOptions>
+{
+    private const string CommandTitle = "List Azure Monitor Metric Namespaces";
+    private readonly ILogger<MetricsNamespacesCommand> _logger = logger;
+
+    private readonly Option<int> _limitOption = MonitorOptionDefinitions.Metrics.NamespacesLimit;
+    private readonly Option<string> _searchStringOption = MonitorOptionDefinitions.Metrics.SearchString;
+
+    public override string Name => "namespaces";
+
+    public override string Description =>
+        $"""
+        List available metric namespaces for an Azure resource. Returns the namespaces that contain metrics for the resource.
+        Required options:
+        - {_resourceNameOption.Name}: {_resourceNameOption.Description}
+        Optional options:
+        - {_resourceGroupOption.Name}: {_resourceGroupOption.Description}
+        - {_resourceTypeOption.Name}: {_resourceTypeOption.Description}
+        - {_limitOption.Name}: {_limitOption.Description}
+        - {_searchStringOption.Name}: {_searchStringOption.Description}
+        """;
+
+    public override string Title => CommandTitle;
+
+    protected override void RegisterOptions(Command command)
+    {
+        base.RegisterOptions(command);
+        command.AddOption(_limitOption);
+        command.AddOption(_searchStringOption);
+    }
+
+    protected override MetricsNamespacesOptions BindOptions(ParseResult parseResult)
+    {
+        var options = base.BindOptions(parseResult);
+        options.Limit = parseResult.GetValueForOption(_limitOption);
+        options.SearchString = parseResult.GetValueForOption(_searchStringOption);
+        return options;
+    }
+
+    [McpServerTool(
+        Destructive = false,
+        ReadOnly = true,
+        Title = CommandTitle)]
+    public override async Task<CommandResponse> ExecuteAsync(CommandContext context, ParseResult parseResult)
+    {
+        var options = BindOptions(parseResult);
+
+        try
+        {
+            // Required validation step
+            if (!Validate(parseResult.CommandResult, context.Response).IsValid)
+            {
+                return context.Response;
+            }
+
+            // Get the metrics service from DI
+            var service = context.GetService<IMonitorMetricsService>();
```

Comment:
Use constructor IoC injection as much as possible vs calling the service collection directly.

Reply:
Same applies to all other commands.

Reply:
This is the standard way other commands do it...I followed convention here. In the services, it uses dependency injection but I believe that command doesn't have the full SP context.